### ISSUE TITLE
RUF055 should categorize `)` as a regex metacharacter

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#15316]
[`unnecessary-regular-expression` (RUF055)](https://docs.astral.sh/ruff/rules/unnecessary-regular-expression/) treats `)` as a normal character that needs no escaping, but it is actually a metacharacter. This leads to false positives and fixes that change program behavior.

```console
$ cat ruf055.py
import re
print(re.split(r")", "a)b"))

$ python ruf055.py 2>&1 | tail -n 1
re.error: unbalanced parenthesis at position 0

$ ruff check --isolated --preview --select RUF055 ruf055.py --fix
Found 1 error (1 fixed, 0 remaining).

$ cat ruf055.py
import re
print("a)b".split(")"))

$ python ruf055.py
['a', 'b']
```

The current behavior was inspired by [this review comment](https://github.com/astral-sh/ruff/pull/14659#discussion_r1862540485), but that comment only applies to `]` and `}`, not `)`.
